### PR TITLE
use handlers from global default

### PR DESCRIPTION
**Pull request by logan-markewich**
Instead of trying to ensure the callback manager is passed into every component, we can actually just override the handlers in the constructor with the handlers in the global settings.

Fixes https://github.com/run-llama/llama_index/issues/13354

### CODE REVIEW

File: llama-index-core/llama_index/core/callbacks/base.py [hunk 1]
```diff
@@ -68,6 +68,15 @@ def __init__(self, handlers: Optional[List[BaseCallbackHandler]] = None):
                     )
             handlers.append(new_handler)
 
+        # if we passed in no handlers, use the global default
+        if len(handlers) == 0:
+            from llama_index.core.settings import Settings
+
+            # hidden var access to prevent recursion in getter
```
Comment (nerdai) [inline llama-index-core/llama_index/core/callbacks/base.py:75]: oh interesting..

**Review by SinghCoder — APPROVED** [review]
should explicit initialization to Settings.callback_manager be removed from other modules' constructors?

File: llama-index-core/llama_index/core/callbacks/base.py [hunk 2]
```diff
@@ -68,6 +68,15 @@ def __init__(self, handlers: Optional[List[BaseCallbackHandler]] = None):
...
             handlers.append(new_handler)
 
+        # if we passed in no handlers, use the global default
+        if len(handlers) == 0:
+            from llama_index.core.settings import Settings
+
+            # hidden var access to prevent recursion in getter
+            cb_manager = Settings._callback_manager
+            if cb_manager is not None:
+                handlers = cb_manager.handlers
```
Comment (SinghCoder) [inline llama-index-core/llama_index/core/callbacks/base.py:78]: is the default value of Settings.callback_manager.handlers an empty list? if not, should we do that else change might break some assumptions downstream of people relying on by default this value to be []

Comment (logan-markewich) [reply llama-index-core/llama_index/core/callbacks/base.py:78]: The default is an empty list yes

**Comment by logan-markewich** [comment]
eh, its fine to keep the other modules construstructors. This is just more of a catch-all. 

The callback manager will be removed hopefully in a few months (replaced by instrumentation)

File: llama-index-core/llama_index/core/callbacks/base.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-core/llama_index/core/callbacks/base.py:75]: Yea just due to how the setter works (its lazily initialized)

**Pull request merged**